### PR TITLE
calculator: added basic calculator operator and now it just work

### PR DESCRIPTION
**Pull request by Supercip971**
for the moment there is 'only' 5 operator : 
add (+)
substract (-)
equal (=)
clear (c)
clear all (ce)﻿

### CODE REVIEW

File: applications/calculator/main.cpp [hunk 1]
```diff
@@ -1,14 +1,172 @@
...
+
+        if (handle_operator(code))
+        {
+
+            enter_switch_operator();
+            current_operator = code;
+        }
+    }
+    update_calculator_text();
+}
```
Comment (sleepy-monax) [inline applications/calculator/main.cpp:135]: This should be a class

File: applications/calculator/calculator.markup [hunk 2]
```diff
@@ -8,15 +8,15 @@
...
 
     <Container layout="grid(4, 6, 4, 4)" fill>
-        <Button icon="percent" outlined/> <Button text="CE" outlined/> <Button text="C" outlined/> <Button icon="backspace-outline" outlined/>
+        <Button icon="percent" outlined/> <Button id="clear_all" text="CE" outlined/> <Button id="clear" text="C" outlined/> <Button icon="backspace-outline" outlined/>
         <Button text="1/x" outlined/> <Button icon="exponent" outlined/> <Button icon="square-root" outlined/> <Button icon="slash-forward" outlined/>
-        <Button text="7"/> <Button text="8" /> <Button text="9" /> <Button icon="close" outlined/>
-        <Button text="4" /> <Button text="5" /> <Button text="6" /> <Button icon="minus" outlined/>
-        <Button text="1" /> <Button text="2" /> <Button text="3" /> <Button icon="plus" outlined/>
-        <Button icon="plus-minus-variant" outlined/> <Button text="0" /> <Button text="." outlined/> <Button icon="equal" filled/>
+        <Button id="7" text="7"/> <Button id="8" text="8" /> <Button id="9" text="9" /> <Button icon="close" outlined/>
```
Comment (sleepy-monax) [inline applications/calculator/calculator.markup:17]: To ensure your code to be future proof, widgets IDs should also be valid C++ identifiers.

**Comment by Supercip971** [comment]
fixed ! ;)

File: applications/calculator/main.cpp [hunk 3]
```diff
@@ -25,148 +25,191 @@ enum CALCULATOR_OPERATOR
...
+
+    Window *target;
+    double main_number = 0;
+    double second_number = 0;
+    int selectionned_number = 0;
+    int current_number_count = 0;
+    double current_operator = 0;
+    bool do_reset_after = true;
+    int calculator_text_length = 0;
+    char calculator_text[32];
```
Comment (sleepy-monax) [inline applications/calculator/main.cpp:55]: Class attributes should be prefixed by a `_` also you forgot to put `private:` before them (I know it useless but it makes the code more consistent)

File: applications/calculator/main.cpp [hunk 4]
```diff
@@ -25,148 +25,191 @@ enum CALCULATOR_OPERATOR
...
+#define CALC_ID_BUTTON_7 "7"
+#define CALC_ID_BUTTON_8 "8"
+#define CALC_ID_BUTTON_9 "9"
+#define CALC_ID_BUTTON_PLUS "plus"
+#define CALC_ID_BUTTON_MINUS "minus"
+#define CALC_ID_BUTTON_EQUAL "equal"
+#define CALC_ID_BUTTON_CLEAR "clear"
+#define CALC_TEXT_BOX_NAME "calcscreen"
+
+class calculator_system
```
Comment (sleepy-monax) [inline applications/calculator/main.cpp:44]: This is not a calculator_system, it's Calculator please call a spade a spade ;)

File: applications/calculator/main.cpp [hunk 5]
```diff
@@ -25,148 +25,191 @@ enum CALCULATOR_OPERATOR
...
+    bool do_reset_after = true;
+    int calculator_text_length = 0;
+    char calculator_text[32];
+
+public:
+    void init_button()
     {
-        format_double(FORMAT_DECIMAL, second_number, (char *)calculator_text, 32);
+
+        CALCULATOR_BUTTON(CALC_ID_BUTTON_0, true, 0)
```
Comment (sleepy-monax) [inline applications/calculator/main.cpp:61]: It's not the Calculator object responsibility to managed windows events. Please respect the separation of concerns.

File: applications/calculator/main.cpp [hunk 6]
```diff
@@ -25,148 +25,191 @@ enum CALCULATOR_OPERATOR
     OPERATOR_CLEAR_ALL = 6
 };
 
-void update_calculator_text()
+#define CALC_ID_BUTTON_0 "0"
```
Comment (sleepy-monax) [inline applications/calculator/main.cpp:28]: This is not what I meant sorry, widget IDs sould be "button0" instead of just "0" for exemple.

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

**Comment by Supercip971** [comment]
(re-fixed)

File: applications/calculator/main.cpp (hunk 4 above)
Comment (sleepy-monax) [reply applications/calculator/main.cpp:52]: @Supercip971 looks like you didn't see this message

Comment (Supercip971) [reply applications/calculator/main.cpp:52]: ow yes sorry

Comment (sleepy-monax) [reply applications/calculator/main.cpp:52]: It's my fault I didn't include this message in the review :sweat_smile:

Comment (Supercip971) [reply applications/calculator/main.cpp]: it is now fixed

**Comment by Supercip971** [comment]
(re-re-fixed)

**Pull request merged**